File: src/0349-Intersection-of-Two-Arrays/intersection_of_two_arrays.py
from typing import List, Set

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1: Set[int] = set(nums1)
        set2: Set[int] = set(nums2)

        ans: List[int] = list(set1 & set2)
        
        return ans

if __name__ == "__main__":
    nums1: List[int] = [1,2,2,1]
    nums2: List[int] = [2,2]
    solution = Solution()

    print(solution.intersection(nums1, nums2))

File: src/0349-Intersection-of-Two-Arrays/intersection_of_two_arrays.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

class Solution {
public:
    std::vector<int> intersection(std::vector<int>& nums1, std::vector<int>& nums2) {
        std::unordered_set<int> set1(nums1.begin(), nums1.end());
        std::unordered_set<int> set2(nums2.begin(), nums2.end());

        std::vector<int> ans;
        for (int num : set1) {
            if (set2.find(num) != set2.end()) {
                ans.push_back(num);
            }
        }

        return ans;
    }
};

int main() {
    std::vector<int> nums1 = {1,2,2,1};
    std::vector<int> nums2 = {2,2};
    Solution solution;

    std::vector<int> output = solution.intersection(nums1, nums2);

    for (auto it : output) {
        std::cout << it << " ";
    }

    return 0;
}